test/nft/anvil/NFT.t.sol

```diff
@@ -138,13 +129,68 @@ contract NFTAnvil is AnvilFork {
     address owner = vault721.ownerOf(vaultId);
     uint256 initBal = vault721.balanceOf(owner);
 
-    address reciever = address(0);
+    address receiver = address(0);
 
     vm.startPrank(owner);
     vm.expectRevert('ERC721: transfer to the zero address');
-    vault721.transferFrom(owner, reciever, vaultId);
+    vault721.transferFrom(owner, receiver, vaultId);
     vm.stopPrank();
 
     assertEq(initBal, vault721.balanceOf(owner));
   }
+
+  /**
+   * @dev Test transfering collateral to an address
+   * that isn't a safeHandler reverts.
+   */
+  function test_revert_If_TransferCollateral_To_NonSafeHandler(
+    uint256 collateral,
+    uint256 cTypeIndex
+  ) public maxLock(collateral) {
+    cTypeIndex = bound(cTypeIndex, 1, cTypes.length - 1); // range: WSTETH, CBETH, RETH, MAGIC
+    address alice = users[0];
+    address proxy = proxies[0]; // alice's proxy
+    bytes32 cType = cTypes[cTypeIndex];
+    uint256 vaultId = _helperDepositCollateralAndGenerateDebt(alice, proxy, cType, collateral, 0);
+
+    vm.startPrank(proxy);
+    vm.expectRevert(IODSafeManager.HandlerDoesNotExist.selector);
+    safeManager.transferCollateral(vaultId, alice, collateral);
+    vm.stopPrank();
+  }
+
+  /**
+   * @dev Test transfering collateral to an address that is a safeHandler
+   * succeeds
+   */
+  function test_transferCollateral_To_SafeHandler(uint256 collateral, uint256 cTypeIndex) public maxLock(collateral) {
+    cTypeIndex = bound(cTypeIndex, 1, cTypes.length - 1); // range: WSTETH, CBETH, RETH, MAGIC
+    address alice = users[0];
+    address aliceProxy = proxies[0]; // alice's proxy
+    bytes32 cType = cTypes[cTypeIndex];
+    uint256 aliceVaultId = _helperDepositCollateralAndGenerateDebt(alice, aliceProxy, cType, 100, 0);
+
+    address bob = users[1];
+    address bobProxy = proxies[1]; // bob's proxy
+    uint256 bobVaultId = vaultIds[bobProxy][cType];
+
+    IODSafeManager.SAFEData memory bobSafeData = safeManager.safeData(bobVaultId);
+    address bobSafeHandler = bobSafeData.safeHandler;
+    assertEq(
+      safeEngine.safes(cType, bobSafeHandler).lockedCollateral,
+      0,
+      'test_transferCollateralToSafeHandler: collateral is empty'
+    );
+
+    vm.startPrank(aliceProxy);
+    // TODO: when we deposit collateral, it is locked, how do we move it from locked to tokenCollateral so 
```

Comment:
This test illustrates that `transferCollateral` works when targeting a safeHandler, but I am still unsure how you move collateral from locked to the `tokenCollateral` storage?

Reply:
#252 tracks this